src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -136,6 +137,29 @@ case class CsvRelation protected[spark] (
     }
   }
 
+  // The function below was borrowed from JSONRelation
+  override def insert(data: DataFrame, overwrite: Boolean) = {
+    val filesystemPath = new Path(location)
+    val fs = filesystemPath.getFileSystem(sqlContext.sparkContext.hadoopConfiguration)
+
+    if (overwrite) {
+      try {
+        fs.delete(filesystemPath, true)
+      } catch {
+        case e: IOException =>
```

Comment:
Wait, it does fail, doesn't it? It catches exception and throws one with a nice message. Or if it's unmatched (unlikely), we will get an even uglier exception :)
Anyway, that's the way JSON does it.